src/server/db/models/CommentModel.ts

```diff
@@ -0,0 +1,18 @@
+import { commentSchema } from "@/utils/types";
+import mongoose, { Schema } from "mongoose";
+import { z } from "zod";
+
+export type Comment = z.infer<typeof commentSchema>;
+
```

Comment:
Oops I didn't know this existed before writing the ticket, but we should also create an input schema to allow parameters with default values to be optional (following this [article](https://stackoverflow.com/questions/72771130/how-to-make-an-optional-property-with-a-default-value-in-zod)).
```suggestion
export type CommentInput = z.input<typeof commentSchema>;

```